realm/realm-library/src/syncIntegrationTest/java/io/realm/objectserver/ManagementRealmTests.java

```diff
@@ -0,0 +1,159 @@
+
+    @Rule
+    public RunInLooperThread looperThread = new RunInLooperThread();
+
+    @Test
+    @RunTestInLooperThread
+    public void create_acceptOffer() {
+        RealmLog.setLevel(LogLevel.DEBUG);
+        SyncUser user1 = UserFactory.createUser(Constants.AUTH_URL, "user1");
+        final SyncUser user2 = UserFactory.createUser(Constants.AUTH_URL, "user2");
+
+        // 1. User1 creates Realm that user2 does not have access
+        final String user1RealmUrl = "realm://127.0.0.1:9080/" + user1.getIdentity() + "/permission-offer-test";
+        SyncConfiguration config1 = new SyncConfiguration.Builder(user1, user1RealmUrl).
+                errorHandler(new SyncSession.ErrorHandler() {
+                    @Override
+                    public void onError(SyncSession session, ObjectServerError error) {
+                        fail("Realm 1 unexpected error: " + error);
+                    }
+                })
+                .build();
+        final Realm realm1 = Realm.getInstance(config1);
+        looperThread.testRealms.add(realm1);
+        realm1.executeTransactionAsync(new Realm.Transaction() {
+            @Override
+            public void execute(Realm realm) {
+                realm.createObject(Dog.class);
+            }
+        });
+
+        // 2. Create configuration for User2's Realm.
+        final SyncConfiguration config2 = new SyncConfiguration.Builder(user2, user1RealmUrl).build();
+
+        // 3. Create PermissionOffer
+        final AtomicReference<String> offerId = new AtomicReference<String>(null);
+        final Realm user1ManagementRealm = user1.getManagementRealm();
+        looperThread.testRealms.add(user1ManagementRealm);
+        user1ManagementRealm.executeTransactionAsync(new Realm.Transaction() {
+            @Override
+            public void execute(Realm realm) {
+                boolean readPermission = true;
+                boolean readWritePermission = true;
+                boolean managePermission = false;
+                Date expiresAt = null;
+                PermissionOffer offer = new PermissionOffer(user1RealmUrl, readPermission, readWritePermission, managePermission, expiresAt);
+                offerId.set(offer.getId());
+                realm.copyToRealm(offer);
+            }
+        }, new Realm.Transaction.OnSuccess() {
+            @Override
+            public void onSuccess() {
+                // 4. Wait for offer to get an token
+                RealmLog.error("OfferID: " + offerId.get());
+                RealmResults<PermissionOffer> offers = user1ManagementRealm.where(PermissionOffer.class)
+                        .equalTo("id", offerId.get())
+                        .findAllAsync();
+                looperThread.keepStrongReference.add(offers);
+                offers.addChangeListener(new RealmChangeListener<RealmResults<PermissionOffer>>() {
+                    @Override
+                    public void onChange(RealmResults<PermissionOffer> offers) {
```

Comment:
This might be easier to read if at least the RealmChangeListener class were pulled up and named.